runtime/runtime-params-estimator/test-contract/src/lib.rs

```diff
@@ -439,6 +442,66 @@ pub unsafe fn keccak512_10kib_10k() {
     }
 }
 
+// Function to measure `ripemd160_base` and `ripemd160_byte`. Also measures `base`, `write_register_base`,
+// and `write_register_byte`. However `ripemd160` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute ripemd160 on 10b 10k times.
+#[no_mangle]
+pub unsafe fn ripemd160_10b_10k() {
+    let buffer = [65u8; 10];
+    for _ in 0..10_000 {
+        ripemd160(buffer.len() as u64, buffer.as_ptr() as *const u64 as u64, 0);
+    }
+}
+// Function to measure `ripemd160_base` and `ripemd160_byte`. Also measures `base`, `write_register_base`,
+// and `write_register_byte`. However `ripemd160` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute ripemd160 on 10kib 10k times.
+#[no_mangle]
+pub unsafe fn ripemd160_10kib_10k() {
+    let buffer = [65u8; 10240];
+    for _ in 0..10_000 {
+        ripemd160(buffer.len() as u64, buffer.as_ptr() as *const u64 as u64, 0);
+    }
+}
+
+// Function to measure `blake2b_base` and `blake2b_byte`. Also measures `base`, `write_register_base`,
+// and `write_register_byte`. However `blake2b` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute blake2b on 10b 10k times.
+#[no_mangle]
+pub unsafe fn blake2b_10b_10k() {
+    let buffer = [65u8; 10];
+    for _ in 0..10_000 {
+        blake2b(buffer.len() as u64, buffer.as_ptr() as *const u64 as u64, 0);
+    }
+}
+// Function to measure `blake2b_base` and `blake2b_byte`. Also measures `base`, `write_register_base`,
+// and `write_register_byte`. However `blake2b` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute blake2b on 10kib 10k times.
+#[no_mangle]
+pub unsafe fn blake2b_10kib_10k() {
+    let buffer = [65u8; 10240];
+    for _ in 0..10_000 {
+        blake2b(buffer.len() as u64, buffer.as_ptr() as *const u64 as u64, 0);
+    }
+}
+
+// Function to measure `ecrecover_base`. Also measures `base`, `write_register_base`, and
+// `write_register_byte`. However `ecrecover` computation is more expensive than register writing
+// so we are okay overcharging it.
+// Compute ecrecover 10k times.
+#[no_mangle]
+pub unsafe fn ecrecover_10k() {
+    let hash = [0u8; 32];
+    let signature = [0u8; 65];
+    let (r, s, v) = (&signature[0..32], &signature[32..64], 27);
```

Comment:
Should this test measure recovering a valid signature

Reply:
Initially I had wanted to do that, but this test contract apparently shouldn't have any dependencies (e.g., on the `hex-literals` crate) so I ripped that code out. I plan to reformulate later using more cumbersome code to initialize the hash and signature word by word.